checkov/terraform/checks/resource/gcp/DataprocPrivateCluster.py

```diff
@@ -0,0 +1,26 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from typing import List
+
+
+class DataprocPrivateCluster(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure that Dataproc clusters are not anonymously or publicly accessible"
+        id = "CKV_GCP_96"
+        supported_resources = ['google_dataproc_cluster_iam_member', 'google_dataproc_cluster_iam_binding']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
```

Comment:
you can actually leverage an attribute, which is populated by the parent class `self.entity_type`. When the check is run you will get the correct resource type.

Reply:
here is an example for a CloudFormation check
https://github.com/bridgecrewio/checkov/blob/9003cc365ff885714b7affbb861db5bca3c203bb/checkov/cloudformation/checks/resource/aws/LambdaDLQConfigured.py#L15-L18

Reply:
Oh neat, I'll play around with that and push something. Thanks!

Reply:
Alright so I switched around the logic a little to create two different paths depending on the resource being evaluated. I think this is quicker and easier to read but I'm open to suggestions.

https://github.com/bridgecrewio/checkov/pull/2450/commits/d855ebbd6146274db58e5e472ca925c69f26b306